week_4/class_rational_part_1/class_rational_part_1.cpp

```diff
@@ -0,0 +1,120 @@
+#include <iostream>
+
+class Rational {
+ public:
+  Rational() {
+    p = 0;
+    q = 1;
+    divider = 1;
+  }
+
+  Rational(int numerator, int denominator) {
+    if (numerator == 0) {
+      p = numerator;
+      q = 1;
```

Comment:
@Ypostolov let's implement copy constructor for that

Reply:
Emplemented.
Please check 6f94025d2712317c65697f93f5dc11e2eb1b0da5

Reply:
@Ypostolov I don't see copy constructor in 6f94025d2712317c65697f93f5dc11e2eb1b0da5
And please, do not squash commits during review which you are referencing

Reply:
@Ypostolov I see that changes in different commit 09c8a4ad275340d60d9684d1a5ac510accdfe17c
So, why this copy constructor do not any copying anything?

Reply:
Fixed in 682be2784f7a4b735014298dac5a1905393ae59a
Please take a look.